sql/src/main/java/io/crate/operation/projectors/ShardDMLExecutor.java

```diff
@@ -137,20 +117,22 @@ private void addRowToRequest(Row row) {
 
     @Override
     public CompletableFuture<? extends Iterable<Row>> apply(BatchIterator<Row> batchIterator) {
+        BatchIterator<TReq> reqBatchIterator =
+            BatchIterators.partition(batchIterator, bulkSize, requestFactory, this::addRowToRequest);
         return new BatchIteratorBackpressureExecutor<>(
-            batchIterator,
             scheduler,
-            rowConsumer,
+            reqBatchIterator,
             this::executeBatch,
-            shouldPause,
-            bulkSize,
+            (a, b) -> a + b,
+            0L,
+            item -> shouldPause.getAsBoolean(),
```

Comment:
Should we use some method references to add some semantics to this constructor call? I think one would have to go back and forth (inside the constructor and back) to figure out what is (a, b) -> a+b.
Also why do we pass an "item" in order to check if we shouldPause?